pkg/antctl/command_definition_test.go

```diff
@@ -439,3 +441,422 @@ func TestCommandDefinitionGenerateExample(t *testing.T) {
+		name     string
+		mode     string
+		cd       *commandDefinition
+		expected bool
+	}{
+		{
+			name:     "Command with no supported component",
+			mode:     "",
+			cd:       &commandDefinition{},
+			expected: false,
+		},
+		{
+			name: "Command for agent defines resource endpoint",
+			mode: "agent",
+			cd: &commandDefinition{
+				agentEndpoint: &endpoint{
+					resourceEndpoint: &resourceEndpoint{
+						namespaced: true,
+					},
+				},
+			},
+			expected: true,
+		},
+		{
+			name: "Command for agent defines non-resource endpoint",
+			mode: "agent",
+			cd: &commandDefinition{
+				agentEndpoint: &endpoint{
+					nonResourceEndpoint: &nonResourceEndpoint{
+						path: "/version",
+					},
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "Command for controller defines resource endpoint",
+			mode: "controller",
+			cd: &commandDefinition{
+				controllerEndpoint: &endpoint{
+					resourceEndpoint: &resourceEndpoint{
+						namespaced: true,
+					},
+				},
+			},
+			expected: true,
+		},
+		{
+			name: "Command for controller defines non-resource endpoint",
+			mode: "controller",
+			cd: &commandDefinition{
+				controllerEndpoint: &endpoint{
+					nonResourceEndpoint: &nonResourceEndpoint{
+						path: "/version",
+					},
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "Command for flow aggregator defines resource endpoint",
+			mode: "flowaggregator",
+			cd: &commandDefinition{
+				flowAggregatorEndpoint: &endpoint{
+					resourceEndpoint: &resourceEndpoint{
+						namespaced: true,
+					},
+				},
+			},
+			expected: true,
+		},
+		{
+			name: "Command for flow aggregator defines non-resource endpoint",
+			mode: "flowaggregator",
+			cd: &commandDefinition{
+				flowAggregatorEndpoint: &endpoint{
+					nonResourceEndpoint: &nonResourceEndpoint{
+						path: "/version",
+					},
+				},
+			},
+			expected: false,
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			runtime.Mode = tt.mode
+			read := tt.cd.namespaced()
+			assert.Equal(t, tt.expected, read)
+		})
+	}
+
+}
+func TestValidate(t *testing.T) {
+	tc := []struct {
+		name     string
+		cd       *commandDefinition
+		expected []string
+		mode     string
+	}{
+		{
+			name: "Command with no name and supported component",
+			cd: &commandDefinition{
+				use: "",
+			},
+			expected: []string{"the command does not have name", ": command does not define output struct", ": command does not define any supported component"},
+		},
+		{
+			name: "Command with name and aliases",
+			cd: &commandDefinition{
+				use:     "controllerinfo",
+				aliases: []string{"controllerinfo", "controllerinfos", "ci", "controllerinfos"},
+			},
+			expected: []string{"controllerinfo: command alias is the same with use of the command", "controllerinfo: command alias is provided twice: controllerinfos", "controllerinfo: command does not define output struct", "controllerinfo: command does not define any supported component"},
```

Comment:
It's too long, you can wrap them into below format. Please fix other occurrences.
```
expected: []string{
"controllerinfo: command alias is the same with use of the command", 
"controllerinfo: command alias is provided twice: controllerinfos",
 "controllerinfo: command does not define output struct", 
"controllerinfo: command does not define any supported component"
},
```

Reply:
Fixed